src/stratigraphy/main.py

```diff
@@ -37,474 +25,68 @@
-        max_coverage = max(coverage_by_generating_line, key=len)
-        description_clusters.append(max_coverage)
-        is_description = [line for line in is_description if line not in max_coverage]
-
-    candidate_rects = []
-
-    for cluster in description_clusters:
-        best_y0 = min([line.rect.y0 for line in cluster])
-        best_y1 = max([line.rect.y1 for line in cluster])
-
-        min_description_x0 = min(
-            [
-                line.rect.x0 - 0.01 * line.rect.width for line in cluster
-            ]  # How did we determine the 0.01? Should it be a parameter? What would it do if we were to change it?
-        )
-        max_description_x0 = max(
-            [
-                line.rect.x0 + 0.2 * line.rect.width for line in cluster
-            ]  # How did we determine the 0.2? Should it be a parameter? What would it do if we were to change it?
-        )
-        good_lines = [
-            line
-            for line in candidate_description
-            if line.rect.y0 >= best_y0 and line.rect.y1 <= best_y1
-            if min_description_x0 < line.rect.x0 < max_description_x0
-        ]
-        best_x0 = min([line.rect.x0 for line in good_lines])
-        best_x1 = max([line.rect.x1 for line in good_lines])
-
-        # expand to include entire last block
-        def is_below(best_x0, best_y1, line):
-            return (
-                (
-                    line.rect.x0 > best_x0 - 5
-                )  # How did we determine the 5? Should it be a parameter? What would it do if we were to change it?
-                and (line.rect.x0 < (best_x0 + best_x1) / 2)  # noqa B023
-                and (
-                    line.rect.y0 < best_y1 + 10
-                )  # How did we determine the 10? Should it be a parameter? What would it do if we were to change it?
-                and (line.rect.y1 > best_y1)
-            )
-
-        continue_search = True
-        while continue_search:
-            line = next((line for line in lines if is_below(best_x0, best_y1, line)), None)
-            if line:
-                best_x0 = min(best_x0, line.rect.x0)
-                best_x1 = max(best_x1, line.rect.x1)
-                best_y1 = line.rect.y1
-            else:
-                continue_search = False
-
-        candidate_rects.append(fitz.Rect(best_x0, best_y0, best_x1, best_y1))
-
-    if len(candidate_rects) == 0:
-        return None
-    if depth_column:
-        return max(candidate_rects, key=lambda rect: score_column_match(depth_column, rect))
-    else:
-        return candidate_rects[0]
-
-
-def perform_matching(directory: Path, **params: dict) -> dict:
-    """Perform the matching of text blocks with depth intervals.
-
+@click.command()
+@click.option(
+    "-i",
+    "--input_directory",
+    type=click.Path(exists=True, path_type=Path),
+    default=DATAPATH / "Benchmark",
+    help="Path to the input directory.",
```

Comment:
Update documentation to clarify that this can also be the path of a single PDF file.